plugins/source/hackernews/codegen/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path"
+	"reflect"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/cloudquery/cloudquery/plugins/source/hackernews/codegen/recipes"
+	"github.com/cloudquery/cloudquery/plugins/source/hackernews/codegen/services"
+	"github.com/cloudquery/cloudquery/plugins/source/hackernews/codegen/tables"
+	"github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+//go:embed templates/*.go.tpl
+var templatesFS embed.FS
+
+func main() {
+	err := services.Generate()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var resources []*recipes.Resource
+	resources = append(resources, recipes.ItemResources()...)
+
+	for _, r := range resources {
+		r.Infer()
```

Comment:
Since you're not doing any parent-child-things you could move this line to the loop below.

Reply:
Removed the codegen directory completely